src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
+            {
+                byte[] buf = new byte[bufferSize];
```

Comment:
I'm wondering if we should try using `getcwd(NULL, 0)` before doing this loop.  Both the Ubuntu (glibc) and OSX manpages say that they provide the Right Answer when that is called (you'd just need to `free()` the result after making a managed string).

The advantage of that plan is no loop on our normal supported systems.  The disadvantages are that you need a SafeMallocHandle to track the return value, and you have a fallback that's unreachable on our officially supported systems.

Just offering it for consideration.